--- dvbplayer.c.orig	2024-10-24 01:35:48.046359743 +0200
+++ dvbplayer.c.NEU8	2024-12-23 22:49:32.534684195 +0100
@@ -261,6 +261,7 @@ private:
   cFrame *playFrame;
   cFrame *dropFrame;
   bool resyncAfterPause;
+  bool resyncAfterSlowForward;
   void TrickSpeed(int Increment);
   void Empty(void);
   bool NextFile(uint16_t FileNumber = 0, off_t FileOffset = -1);
@@ -317,6 +318,7 @@ cDvbPlayer::cDvbPlayer(const char *FileN
   playFrame = NULL;
   dropFrame = NULL;
   resyncAfterPause = false;
+  resyncAfterSlowForward = false;
   isyslog("replay %s", FileName);
   fileName = new cFileName(FileName, false, false, isPesRecording);
   replayFile = fileName->Open();
@@ -749,12 +751,23 @@ void cDvbPlayer::Play(void)
      DevicePlay();
      playMode = pmPlay;
      playDir = pdForward;
-     if (resyncAfterPause) {
+     if (!resyncAfterSlowForward) {
+        if (resyncAfterPause) {
+           int Current, Total;
+           if (GetIndex(Current, Total, true))
+               Goto(Current);
+           }
+        }
+     else {
         int Current, Total;
-        if (GetIndex(Current, Total, true))
-           Goto(Current);
-        resyncAfterPause = false;
+        if (GetIndex(Current, Total, false)) {
+           int Index = index->GetNextIFrame(Current, true, NULL, NULL, NULL);
+           Goto(Index);
+           }
+        resyncAfterSlowForward = false;
         }
+     if (resyncAfterPause)
+        resyncAfterPause = false;
      }
 }
 
@@ -797,6 +810,7 @@ void cDvbPlayer::Forward(void)
             // run into pmPause
        case pmStill:
        case pmPause:
+            resyncAfterSlowForward = true;
             DeviceMute();
             playMode = pmSlow;
             playDir = pdForward;
@@ -845,7 +859,9 @@ void cDvbPlayer::Backward(void)
             Empty();
             // run into pmPause
        case pmStill:
-       case pmPause: {
+       case pmPause:
+            resyncAfterSlowForward = false;
+            {
             LOCK_THREAD;
             Empty();
             DeviceMute();
